app/data/Owners.scala

```diff
@@ -0,0 +1,37 @@
+package data
+
+import model._
+
+trait Owners {
+  def stacks: Set[(String, SSA)]
+
+  def all: Set[Owner] = stacks
+    .groupBy(_._1)
+    .map { case (ownerId, stacks) =>
+      Owner(ownerId, stacks.map(_._2))
+    }.toSet
+
+  def find(id: String): Option[Owner] = all.find(_.id == id)
+
+  def find(ssa: SSA): Option[Owner] = all.find(_.hasSSA(ssa))
+
+  def forStack(stackName: Option[String], stageName: Option[String], appName: Option[String]): Option[Owner] = all
+    .find(_.hasSSA(SSA(stackName, stageName, appName)))
+    .orElse(all.find(_.hasSSA(SSA(stackName, stageName))))
+    .orElse(all.find(_.hasSSA(SSA(stackName))))
+    .orElse(all.find(_.hasSSA(SSA())))
+}
+
+object Owners extends Owners {
+
+  override def stacks: Set[(String, SSA)] = Set(
+    "phil.wills" -> SSA(), // Default owner of all stacks
```

Comment:
you are right about what kind of requests would be more often executed. However storage can be done differently. I mostly though about people modifying this file: I found it easier to add/find a team and add/modify the stacks that the other way around, but that's a personal opinion.